pkg/apis/core/validation/conditional_validation.go

```diff
@@ -226,3 +228,43 @@ func podSCTPFields(podSpec *api.PodSpec, fldPath *field.Path) []*field.Path {
 	}
 	return fields
 }
+
+// validateMixedProtocolLBSVC checks if the old Service has type=LoadBalancer and whether the Service has different Protocols
+// on its ports. If the MixedProtocolLBSVC feature flag is disabled the usage of different Protocols in the new Service is
+// valid only if the old Service has different Protocols, too.
+func validateMixedProtocolLBSVC(service, oldService *api.Service) (errs field.ErrorList) {
+	if service.Spec.Type != api.ServiceTypeLoadBalancer {
+		return
+	}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.MixedProtocolLBSVC) {
+		if serviceHasMixedProtocols(service) && !serviceHasMixedProtocols(oldService) {
+			errs = append(errs, field.Invalid(field.NewPath("spec.ports"), service.Spec.Ports, "cannot create an external load balancer with mix protocols"))
+		}
+	}
+	return
+}
+
+func serviceHasMixedProtocols(service *api.Service) bool {
+	if service == nil {
+		return false
+	}
+	var protocol api.Protocol = "init"
+	for _, port := range service.Spec.Ports {
+		if protocol == "init" {
+			if port.Protocol == "" {
```

Comment:
You should not need to handle "" - that will be defaulted

Reply:
This whole thing should be as easy as:

```
protos := map[string]bool{}
for _, port := range service.Spec.Ports {
    protos[string(port.Protocol)] = true
}
return len(protos) > 1
```

Reply:
Yeah. I guess, next time I should just leave the implementation of stuff to others better in this.